content/gsoc/2022.md

```diff
@@ -0,0 +1,600 @@
+Adding uplifting for new architectures that weren't even supported by ESIL will imrove the
+analysis to even greater degree.
+
+### Assess requirements for midterm/final evaluation
+- 1st term: finish the RZIL uplifting for the chosen architecture with basic instruction-level tests
+- Final term: implement all changes in the analysis code, added more complex integration tests with types analysis
+
+### Mentors
+- xvilka
+- thestr4ng3r
+
+### Links/Resources
+- [RZIL-labeled issues](https://github.com/rizinorg/rizin/labels/RZIL)
+- [ESIL-labeled issues](https://github.com/rizinorg/rizin/labels/RZIL)
+- [ESIL to RZIL conversion tracking issue](https://github.com/rizinorg/rizin/issues/2080)
+- [Cutter: IL output and graph visual representation](https://github.com/rizinorg/rizin/issues/1133)
+
+## Debug information handling improvements (Medium project)
+
+Rizin already supports most of the DWARF and PDB features, including cross-platform parsing of both.
+But there are huge chunks are missing in terms of the integration an source line information.
+Current task would be to improve parsing support of both, add compatibility with `debuginfod`,
+and enhance breakpoint integration, and variable/structure printing in the debugging mode
+with the source line information.
+
+### Task
+1. Support loading DWARF information from separate files and `debuginfod`
+2. Unify source lines/types information access for DWARF, PDB, dSYM
+3. Integrate source line and types/variables information with the analysis
+4. Integrate source line and types/variables with printing with `p` commands in the debug mode
+5. Integrate source line and types/variables with breakpoint commands and APIs
+6. Parsing performance improvements
+
+### Skills
+Participant should know C as well as have the experience of debugging with GDB or LLDB.
+
+### Difficulty
+Hard
+
+### Benefits for the participant
+Participant will understand how high-level features of debuggers work as well as gain skills in the field of software architecture of a large, modular C project.
+
+### Assess requirements for midterm/final evaluation
+- 1st term: `debuginfod` and source line information refactoring are implemented
+- Final term: Integration of variable information with the analysis and debug printing is implemented
+
+### Mentors
+ - xvilka
+ - thestr4ng3r
+ - ret2libc
+
+### Links/Resources
+- [Loading debug information from `debuginfod`](https://github.com/rizinorg/rizin/issues/1285)
+- [Unify code of source information access for DWARF, PDB, dSYM](https://github.com/rizinorg/rizin/issues/907)
+- [Debuginfod](https://sourceware.org/elfutils/Debuginfod)
+- [Debian Debuginfod](https://wiki.debian.org/Debuginfod)
+- [Fedora Debuginfod](https://fedoraproject.org/wiki/Debuginfod)
+- [DWARF-labeled issues](https://github.com/rizinorg/rizin/labels/DWARF)
+- [PDB-labaled issues](https://github.com/rizinorg/rizin/labels/PDB)
+
+## Thread-safety and multithreading (Medium project)
```

Comment:
Not a big fan of this project... I think we may have better things to propose :(

Reply:
Please share your ideas then.